File: src/data/courses/dsa/greedy/others/jump-game/code/python/jump-game.py
def can_jump(nums):
    max_reach = 0  # Maximum index we can reach
    for i in range(len(nums)):
        if i > max_reach:
            return False  # Can't reach this index
        max_reach = max(max_reach, i + nums[i])  # Update max reach
    return True

File: src/data/courses/dsa/greedy/others/jump-game/code/java/JumpGame.java
class JumpGame {
    public boolean canJump(int[] nums) {
        int maxReach = 0; // Maximum index we can reach
        for (int i = 0; i < nums.length; i++) {
            if (i > maxReach) return false; // Can't reach this index
            maxReach = Math.max(maxReach, i + nums[i]); // Update max reach
        }
        return true;
    }
}